c++/src/Utils.hh

```diff
@@ -60,16 +60,27 @@ public:
   }
 };
 
-#define SCOPED_STOPWATCH(METRICS_PTR, LATENCY_VAR, COUNT_VAR)         \
-  AutoStopwatch measure(                                              \
-    (METRICS_PTR == nullptr ? nullptr : &METRICS_PTR->LATENCY_VAR),   \
+#if ENABLE_METRICS
+#define SCOPED_STOPWATCH(METRICS_PTR, LATENCY_VAR, COUNT_VAR)               \
+  AutoStopwatch measure(                                                    \
+    (METRICS_PTR == nullptr ? nullptr : &METRICS_PTR->LATENCY_VAR),         \
     (METRICS_PTR == nullptr ? nullptr : &METRICS_PTR->COUNT_VAR))
 
-#define SCOPED_MINUS_STOPWATCH(METRICS_PTR, LATENCY_VAR)              \
-  AutoStopwatch measure(                                              \
-    (METRICS_PTR == nullptr ? nullptr : &METRICS_PTR->LATENCY_VAR),   \
+#define SCOPED_MINUS_STOPWATCH(METRICS_PTR, LATENCY_VAR)                    \
+  AutoStopwatch measure(                                                    \
+    (METRICS_PTR == nullptr ? nullptr : &METRICS_PTR->LATENCY_VAR),         \
     nullptr, true)
+#else
+#define SCOPED_STOPWATCH(METRICS_PTR, LATENCY_VAR, COUNT_VAR)               \
```

Comment:
If we define the macro as below, the change would be trivial. We can always assign metrics to getDefaultReaderMetrics() and it will not get updated if the macro is disabled.

`#define SCOPED_STOPWATCH(METRICS_PTR, LATENCY_VAR, COUNT_VAR) `
`#define SCOPED_MINUS_STOPWATCH(METRICS_PTR, LATENCY_VAR)`